### PR TITLE
Add support for spark-submit packages and repositories

### PR DESCRIPTION
**Pull request by mdelaney**
- added support for packages, exclude-packages, and repositories
  spark-submit parameters
- added these new parameters to the user guide documentation

### CODE REVIEW

**Comment by mdelaney** [comment]
@googlebot I signed it!

File: pkg/controller/sparkapplication/submission.go [hunk 1]
```diff
@@ -233,6 +233,17 @@ func addDependenciesConfOptions(app *v1beta2.SparkApplication) []string {
 	if len(app.Spec.Deps.PyFiles) > 0 {
 		depsConfOptions = append(depsConfOptions, "--py-files", strings.Join(app.Spec.Deps.PyFiles, ","))
 	}
+	if len(app.Spec.Deps.Packages) > 0 {
+		depsConfOptions = append(depsConfOptions, "--packages", strings.Join(app.Spec.Deps.Packages, ","))
+	}
+	if len(app.Spec.Deps.ExcludePackages) > 0 {
+		depsConfOptions = append(depsConfOptions,
+			"--exclude-packages", strings.Join(app.Spec.Deps.ExcludePackages, ","))
```
Comment (liyinan926) [inline pkg/controller/sparkapplication/submission.go:241]: Nit: there's no need to start a new line here and below.

**Comment by mdelaney** [comment]
Addressed comments, thanks for the super fast review! :-)

File: pkg/apis/sparkoperator.k8s.io/v1beta2/types.go [hunk 2]
```diff
@@ -408,6 +408,20 @@ type Dependencies struct {
 	// PyFiles is a list of Python files the Spark application depends on.
 	// +optional
 	PyFiles []string `json:"pyFiles,omitempty"`
+	// Packages is a list of maven coordinates of jars to include on the driver and executor
+	// classpaths. This will search the local maven repo, then maven central and any additional
+	// remote repositories given by the "repositories" option.
+	// Each papckage should be of the form "groupId:artifactId:version"
```
Comment (liyinan926) [inline pkg/apis/sparkoperator.k8s.io/v1beta2/types.go:414]: Nit: missing period at the end.

File: pkg/apis/sparkoperator.k8s.io/v1beta2/types.go [hunk 3]
```diff
@@ -408,6 +408,20 @@ type Dependencies struct {
...
+	// +optional
+	Packages []string `json:"packages,omitempty"`
+	// ExcludePackages is a list of groupId:artifactId, to exclude while resolving the
```
Comment (liyinan926) [inline pkg/apis/sparkoperator.k8s.io/v1beta2/types.go:417]: Double quotes around `groupId:artifactId`.

**Review by liyinan926 — APPROVED** [review]

**Pull request merged**